src/fondant/core/manifest.py

```diff
@@ -196,13 +143,41 @@ def copy(self) -> "Manifest":
     def metadata(self) -> t.Dict[str, t.Any]:
         return self._specification["metadata"]
 
+    @property
+    def index(self) -> t.Dict[str, t.Any]:
+        return self._specification["index"]
+
     def update_metadata(self, key: str, value: t.Any) -> None:
         self.metadata[key] = value
 
     @property
     def base_path(self) -> str:
         return self.metadata["base_path"]
 
+    @property
+    def field_mapping(self):
+        """
+        Retrieve a mapping of field locations to corresponding field names.
+        A dictionary where keys are field locations and values are lists
+        of column names.
+
+        Example:
+           {
+               "/base_path/component_1": ["Name", "HP"],
+               "/base_path/component_2": ["Type 1", "Type 2"],
+           }
+        """
+        field_mapping = {}
+        for field_name, field in self.fields.items():
+            location = (
+                f"{self.base_path}/{self.pipeline_name}/{self.run_id}{field.location}"
```

Comment:
```suggestion
                f"{self.base_path}/{self.pipeline_name}/{self.run_id}/{field.location}"
```

Is this slash not needed?

Reply:
The schema requires that the location starts with a `/`. What means `field.location` starts always with a `/`.
Look a bit unusual here but it is indeed not needed.
We could think about defining the location without the leading `/`. Not sure if there was a reason to have it.

Reply:
Ok then it's fine. I think it looks clean starting with a `/` indeed.